### PR TITLE
ci: add broken link checker action using lychee

### PR DESCRIPTION
**Pull request by phanirithvij**
- **gitignore .direnv**
- **ci: add broken link checker action**

see example run https://github.com/phanirithvij/nixos-wiki-infra/actions/runs/10896656762

### CODE REVIEW

File: formatter.nix [hunk 1]
```diff
@@ -29,7 +29,6 @@
         ];
         programs.shellcheck.enable = true;
         programs.deno.enable = true;
-        programs.black.enable = true;
```
Comment (phanirithvij) [inline formatter.nix]: oh, I didn't know ruff was a python formatter.

**Comment by Mic92** [comment]
> maybe we could add cachix/git-hooks.nix?
> 
> as an example I have this in git-hooks check in my personal flake
> 
> ```nix
> hooks = {
>     nix-fmt = {
>         enable = true;
>         always_run = true;
>         stages = [ "pre-commit" ];
>         entry = toString (pkgs.writeShellScript "nix-fmt" "nix fmt");
>     };
> };
> ```

Mhm. I made bad experience with `pre-commit` hooks in the past. They are invasive i.e. breaking interactive rebase. If you make the hook opt-in that would be fine with me. I usually just nix fmt and git-absorb after the fact.

**Comment by Mic92** [comment]
https://github.com/NixOS/nixos-wiki-infra/actions/workflows/check-dead-links.yml

**Comment by phanirithvij** [comment]
Ok I didn't know about git-absorb, I will check it out.

**Comment by Mic92** [comment]
Wouldn't it make sense to upload the csv only instead of all the other files?
Than one can just get broken links in one click?

**Comment by phanirithvij** [comment]
There's an issue `"python3.withPackages (ps: [ ps.lxml ])"` works in the shebang the current one doesn't run locally for me.
```nix
            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ ('
```
double quote instead of single

**Comment by phanirithvij** [comment]
> Wouldn't it make sense to upload the csv only instead of all the other files? Than one can just get broken links in one click?

Yea but it's just 6kb (the whole thing combined), I am printing the csv file anyway so it will be in the workflow run logs.

It is just for downloading locally and helpful for navigation in your fav editor (especially the filtered dump xml, I use it to find the article exact location).

**Comment by Mic92** [comment]
https://github.com/NixOS/nixos-wiki-infra/pull/141

**Comment by phanirithvij** [comment]
> Ok. For me the double quote also worked.

Let me be clear, single quote did not work for me, but double quote did. So it sould be fixed.

**Comment by phanirithvij** [comment]
> Wouldn't it make sense to upload the csv only instead of all the other files? Than one can just get broken links in one click?
There's also archive.org suggestions and the full report incase it could be useful to add/remove entries to `allowed.links` file.

**Comment by phanirithvij** [comment]
Also adding here for reference https://github.com/lycheeverse/lychee/issues/1501 needs to be fixed.

**Comment by phanirithvij** [comment]
Feel free to upload just the csv, I see no harm in having the extra files since github actions have a generous 90 day retention and the extra files are slightly useful.

**Pull request merged**